lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -486,6 +486,43 @@ _LoadTexture(const std::string& path, bool& isColorSpaceSRGB, MFloatArray& uvSca
         desc.fFormat = MHWRender::kR32G32B32_FLOAT;
         texture = textureMgr->acquireTexture(path.c_str(), desc, spec.data);
         break;
+    case HioFormatFloat16Vec3: {
+        // R16G16B16 is not supported by VP2. Converted to R16G16B16A16.
+        constexpr int bpp_8 = 8;
+
+        desc.fFormat = MHWRender::kR16G16B16A16_FLOAT;
+        desc.fBytesPerRow = spec.width * bpp_8;
+        desc.fBytesPerSlice = desc.fBytesPerRow * spec.height;
+
+        GfHalf               opaqueAlpha(1.0f);
+        const unsigned short alphaBits = opaqueAlpha.bits();
+        const unsigned char  lowAlpha = reinterpret_cast<const unsigned char*>(&alphaBits)[0];
+        const unsigned char  highAlpha = reinterpret_cast<const unsigned char*>(&alphaBits)[1];
+
+        std::vector<unsigned char> texels(desc.fBytesPerSlice);
+
+        for (int y = 0; y < spec.height; y++) {
+            for (int x = 0; x < spec.width; x++) {
+                const int t = spec.width * y + x;
+                texels[t * bpp_8 + 0] = storage[t * bpp + 0];
+                texels[t * bpp_8 + 1] = storage[t * bpp + 1];
+                texels[t * bpp_8 + 2] = storage[t * bpp + 2];
+                texels[t * bpp_8 + 3] = storage[t * bpp + 3];
+                texels[t * bpp_8 + 4] = storage[t * bpp + 4];
+                texels[t * bpp_8 + 5] = storage[t * bpp + 5];
+                texels[t * bpp_8 + 6] = lowAlpha;
+                texels[t * bpp_8 + 7] = highAlpha;
+            }
+        }
```

Comment:
For scenes with lots of large textures (e.g. 4k textures), this loop could be a hotspot and can be converted into a concurrent loop.

Reply:
@kxl-adsk Are we required to support USD 20.08 and before? If so, the other conditional compilation block needs to have the same change as well.

Reply:
Honestly, given that I would not be able to test it (since Maya does not currently use that old USD version), I would rather not modify code that I would be modifying blindly.

Reply:
We can build with a custom USD version on our own. But let's wait for @kxl-adsk to confirm the requirement first.

Reply:
Yes. Our minimal supported version is 20.02.

Reply:
I just realized that our version of USD 20.02 is *not* built with openimageio... (after all, adding it to USD is what this task is about...) so it cannot possibly load EXR textures.

Reply:
@seando-adsk Do you have any suggestion for Pierre?

Reply:
Oh, I ended-up rebuilding USD 20.02 by hand, enabling oiio and openEXR. It just took me the afternoon to find the correct recipe to get it to compile successfully.